docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+      final (callback) {
+        console.log(`Response pipelined to writable: ${res}`)
+        callback()
+      }
+    }),
+    error => {
+      if (error) {
+        console.error(error)
+      }
+
+      client.close()
+      server.close()
+    }
+  )
+})
+```
+
+### `Client.request(options [, callback])`
+
+Performs a HTTP request.
+
+Non-idempotent requests will not be pipelined in order
+to avoid indirect failures.
+
+Idempotent requests will be automatically retried if
+they fail due to indirect failure from the request
+at the head of the pipeline. This does not apply to
+idempotent requests with a stream request body.
+
+Arguments:
+
+* **options** `RequestOptions`
+* **callback** `(error: Error | null, data: ResponseData) => void` (optional)
+
+Returns: `void | Promise<ResponseData>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Parameter: `RequestOptions`
+
+Extends: [`DispatchOptions`](#parameter-dispatchoptions)
+
+* **opaque** `unknown` (optional) - Default: `null` - Used for passing through context to `ResponseData`
+* **signal** `AbortSignal | events.EventEmitter | null` (optional) - Default: `null`
+
+The `RequestOptions.method` property should not be value `'CONNECT'`.
+
+#### Parameter: `ResponseData`
+
+* **statusCode** `number`
+* **headers** `http.IncomingHttpHeaders`
+* **body** `stream.Readable`
+* **trailers** `Record<string, string>` - This object starts out
+  as empty and will be mutated to contain trailers after `body` has emitted `'end'`.
+* **opaque** `unknown`
+
+#### Example 1 - Basic GET Request
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+
+const server = createServer((request, response) => {
+  response.end('Hello, World!')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  client.request({
+    path: '/',
+    method: 'GET'
+  }).then(({ body, headesr, statusCode, trailers }) => {
+    console.log(`response received ${statusCode}`)
+    console.log('headers', headers)
+    body.setEncoding('utf8')
+    body.on('data', console.log)
+    body.on('end', () => {
+      console.log('trailers', trailers)
+    })
+
+    client.close()
+    server.close()
+  }).catch(error => {
+    console.error(error)
+  })
+})
+```
+
+#### Example 2 - Aborting a request
+
+> Node.js v15+ is required to run this example
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+
+const server = createServer((request, response) => {
+  response.end('Hello, World!')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+  const abortController = new AbortController()
+
+  client.request({
+    path: '/',
+    method: 'GET',
+    signal: abortController.signal
+  }).catch(error => {
+    console.error(error) // should print an RequestAbortedError
+    client.close()
+    server.close()
+  })
```

Comment:
I think it's better to use the callback API in the examples that don't use async/await.

Reply:
Okay good point - I'd prefer async-await, but `server.listen` only supports that callback (since I believe its actually assigned to the server as an event listener for `server.listen`. 

I don't know off the top of my head if an event listener can be an `async` function. I'm going to research that next and make changes accordingly.

Reply:
https://nodejs.org/api/events.html#events_capture_rejections_of_promises

It seems like its a bit dangerous to use async with event listeners ☹️